kolibri/core/assets/src/views/CorePage/AppBarPage.vue

```diff
@@ -150,13 +166,30 @@
       this.$nextTick(() => {
         this.appBarHeight = this.$refs.appBar.$el.scrollHeight || 0;
       });
+      window.addEventListener('scroll', this.handleScroll);
+    },
+    beforeUnmount() {
+      window.removeEventListener('scroll', this.handleScroll);
     },
     methods: {
       findFirstEl() {
         this.$nextTick(() => {
           this.$refs.sideNav.focusFirstEl();
         });
       },
+      handleScroll() {
+        const scrollTop = window.scrollY;
+        //Is user scrolling up?
+        if (scrollTop > this.lastScrollTop) {
+          this.hideAppBars = false;
+        } else {
+          this.hideAppBars = true;
+        }
+
+        setTimeout(() => {
```

Comment:
Just for my understanding, what was the situation that `setTimeout` helped with?

Reply:
Ah I see. Yes, I think we won't need it then. Standard throttling should be more efficient. It seems to me that `setTimeout` placed here won't prevent the logic above from being run.